static/chat.js

```diff
@@ -157,8 +164,10 @@ function receiveReplica(inputs) {
         }
       }
     } else {
-      $('.loading-animation, .speed, .suggest-join').remove();
+      $('.loading-animation, .speed, .suggest-join, .generation-controls').remove();
+      ws.close();
```

Comment:
```suggestion
      resetSession();
```

This is a smarter function that closes the websocket only if it's still open and resets the generation state (so that we feed the prefix to the model if you continue talking to it).